scipy/stats/_continuous_distns.py

```diff
@@ -3616,6 +3616,50 @@ def _cdf(self, x, mu):
     def _stats(self, mu):
         return mu, mu**3.0, 3*np.sqrt(mu), 15*mu
 
+    def fit(self, data, *args, **kwds):
+
+        if type(self) == wald_gen:
+            return super(invgauss_gen, self).fit(data, *args, **kwds)
+
+        floc = kwds.pop('floc', None)
+        fscale = kwds.pop('fscale', None)
+        # borrowed from distn_infrastructure
+        # puts shape into kwds['f0']
+        if self.shapes:
+            shapes = self.shapes.replace(',', ' ').split()
+            for j, s in enumerate(shapes):
+                key = 'f' + str(j)
+                names = [key, 'f' + s, 'fix_' + s]
+                val = _get_fixed_fit_value(kwds, names)
+                if val is not None:
+                    kwds[key] = val
+        f0_s = kwds.pop('f0', None)
+        data = np.asarray(data)
+        _check_fit_input_parameters(data, args, kwds, fixed_param=(floc, fscale, f0_s))
+        '''
+        MLE is not used in 3 condtions:
+        - `floc` is not set
+        - `floc` is set but translation results in negative data
+        - `f0` is fixed
+        These three cases fall back on generic optimization.
+        '''
+        if floc is None or f0_s is not None or np.any(data - floc < 0):
+            if fscale is not None:
+                kwds['fscale'] = fscale
+            if floc is not None:
+                kwds['floc'] = floc
+            if f0_s is not None:
+                kwds['f0'] = f0_s
+            return super(invgauss_gen, self).fit(data, *args, **kwds)
+        else:
+            if floc != 0:
+                data = data - floc
+            f0 = np.mean(data)
+            if fscale is None:
+                fscale = len(data) / (np.sum(data**(-1) - f0**(-1)))
+            f0_s = f0/fscale
```

Comment:
Again we have a case where the optimal scale and optimal location are independent? If so, should explain in comments.